### PR TITLE
feature-RecipeDomain initial

### PR DESCRIPTION
**Pull request by Dissurender**
Implements the Recipe Domain.
Includes:
- Controller
- Entity
- Repo
- Service
- ServiceImpl

## These files still require functionality

Complete issues #60 #61 #62 #63 before merging this PR

### CODE REVIEW

File: Backend_Application/src/main/java/dsd/cohort/application/recipe/RecipeRepository.java
Comment (wcouturiaux) [inline Backend_Application/src/main/java/dsd/cohort/application/recipe/RecipeRepository.java:1]: Missing the @Repository annotation

Comment (Dissurender) [reply Backend_Application/src/main/java/dsd/cohort/application/recipe/RecipeRepository.java:1]: Gotcha, thanks for the call out!

**Comment by Dissurender** [comment]
Putting up for review to merge with ingredient domain changes

File: Frontend_Application/package-lock.json
Comment (Dissurender) [inline Frontend_Application/package-lock.json:1]: Pre-import fix on main

**Review by Dissurender — COMMENTED** [review]
Self-review to verify overwrites

**Pull request merged**